x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecksTests.java

```diff
@@ -60,4 +62,48 @@ public void testCheckProcessors() {
         assertSettingDeprecationsAndWarnings(new Setting<?>[]{EsExecutors.PROCESSORS_SETTING});
     }
 
+    public void testCheckMissingRealmOrders() {
```

Comment:
I'd like an inverse test as well - that realms with an order don't trigger an issue.

Reply:
Added one test for correct realm order configs (2 realms of different orders)